components/admin_console/brand_image_setting/brand_image_setting.tsx

```diff
@@ -16,41 +15,54 @@ import OverlayTrigger from 'components/overlay_trigger';
 
 const HTTP_STATUS_OK = 200;
 
-export default class BrandImageSetting extends React.PureComponent {
-    static propTypes = {
+type Props = {
 
-        /*
-         * Set for testing purpose
-         */
-        id: PropTypes.string,
+    /*
+   * Set for testing purpose
+   */
+    id?: string;
 
-        /*
-         * Set to disable the setting
-         */
-        disabled: PropTypes.bool.isRequired,
+    /*
+   * Set to disable the setting
+   */
+    disabled: boolean;
 
-        /*
-        * Set the save needed in the admin schema settings to trigger the save button to turn on
-        */
-        setSaveNeeded: PropTypes.func.isRequired,
+    /*
+   * Set the save needed in the admin schema settings to trigger the save button to turn on
+   */
+    setSaveNeeded: () => void;
 
-        /*
-        * Registers the function suppose to be run when the save button is pressed
-        */
-        registerSaveAction: PropTypes.func.isRequired,
+    /*
+   * Registers the function suppose to be run when the save button is pressed
+   */
+    registerSaveAction: (saveAction: () => Promise<unknown>) => void;
 
-        /*
-        * Unregisters the function on unmount of the component suppose to be run when the save button is pressed
-        */
-        unRegisterSaveAction: PropTypes.func.isRequired,
-    }
+    /*
+   * Unregisters the function on unmount of the component suppose to be run when the save button is pressed
+   */
+    unRegisterSaveAction: (saveAction: () => Promise<unknown>) => void;
+};
+
+type State = {
+    deleteBrandImage: boolean;
+    brandImage?: Blob;
+    brandImageExists: boolean;
+    brandImageTimestamp: number;
+    error: string;
+};
+
+export default class BrandImageSetting extends React.PureComponent<
+Props,
+State
+> {
```

Comment:
nit: Please update as one-liner, thanks
```suggestion
export default class BrandImageSetting extends React.PureComponent<Props, State> {
```